### ISSUE TITLE
"This page not found" error after upgrading to ownCloud 8.1.1.7

### ISSUE THREAD

**Issue by ThomasDaheim** [SynoCommunity/spksrc#1826]
Hi,

first of all thanks for your awesome work to bring all those packages to Synology!

I have made the upgrade to 8.1.1.7 via the package center (DS213air, DSM 5.2-5592 Update 3) from 8.0.
When trying to logon to ownCloud afterwards it showed the update screen and I confirmed it should update. After completion I end up with "This page not found" when trying to access ownCloud.
Since index.html, index.php, are in the web/owncloud folder I would think it might be a problem with permissions. But all are set to "User: http, Group: root"?
Any hints where I should dig further?

Thanks and sorry if that was already raised previously - didn't find anything,
Thomas

**Comment by ThomasDaheim** [comment]
More info:

I can see that only the "config" folder and a few files have a new timestamp. And of course, the data folder has been emptied...

**Comment by Dr-Bean** [comment]
I'm a bit concerned that you mention that the data folder has been emptied...The data dir may exist, but it's not necessarily filled...unless you have actively chosen that location to store your data, and that's a bad idea.
Your data should be stored safe somewhere in a shared folder on one of the volumes, e.g. the default `/volume1/owncloud`. `/volume1/web` is specifically for web frontends, not data storage.
If you did store it in `/volume1/web`, then it's time to break out the backups, because that data is lost. If you don't have backups, then turn off the DS, and start looking at data restoration.

With that out of the way, I don't know why your Owncloud installation doesn't work. After the package update, Owncloud should be able to update itself just fine. From what you're saying, something went wrong during that process.

You can probably do several things to repair the Owncloud installation. Probably one of the easiest things to do is to replace the contents in `/volume1/web/owncloud`, by downloading and unzipping the file found at https://owncloud.org/install/#instructions-server.
If that does not work, you can remove and reinstall the SynoCommunity package. During removal, you can make a backup of the Owncloud database, and if you're familiar with databases, restore that too. Otherwise, you'll be starting completely from scratch wrt accounts, indexes etc.

**Comment by Dr-Bean** [comment]
Btw, I do assume you have not made changes to the way Owncloud is accessed, or to Webserver in general? You use http://ip-of-nas/owncloud, which results in a 404?

**Comment by ThomasDaheim** [comment]
Hi Dr-Bean,

some more info after I managed to restore my backup (40GB)...

- Not sure if I ever decided that explicitly but all ownCloud user directories are under /volume1/web/owncloud. How could I change that to /volume1/owncloud to avoid data loss if a backup fails?

- Yes, my owncloud instance is accessed via http://ip-of-nas/owncloud and I get the 404 from there

- I have now both owncloud instances (8.0 and 8.1) in separate folders and will see if I can spot the difference...

Will keep you posted on the progress,
Thomas

**Comment by Dr-Bean** [comment]
The data location is chosen during the package installation. I would indeed suggest to change the location, otherwise you'll see this happening with every update.

Changing the data dir in a running Owncloud is somewhat supported, at least according to the following link: https://forum.owncloud.org/viewtopic.php?f=17&t=7118. Looks easy enough, but that might depend on your familiarity with Linux environments. Make sure the http user can access the data directory.

As to the 404: it's reasonably simple to set up a frontend for Owncloud. Take a look at our installer for instance (https://github.com/SynoCommunity/spksrc/blob/master/spk/owncloud/src/installer.sh#L55-L76).
In short, for testing purposes, you can create a folder such as `/volume1/web/owncloudtest`, and unzip the Owncloud files there. Without doing anything else, open http://ip-of-nas/owncloudtest, and you should see the initial Owncloud setup page with that.

**Comment by ThomasDaheim** [comment]
OK, some progress :-)

- Managed to change my data directory to some other place

- Unzipped owncloud 8.1 in a new directory <code>/volume1/web/owncloud81</code> and I end up in the initial setup dialogue

So my next step would be to copy things from the <code>/volume1/web/owncloud81</code> to <code>/volume1/web/owncloud</code>. But I'm sure I shouldn't simply overwrite all files, right? But on the other hand that seems to be what <code>cp -pR ${INSTALL_DIR}/share/${PACKAGE} ${WEB_DIR}</code> from your installer is doing? Besides saving / restoring the config.php file in <code>preupgrade() / postupgrade()</code>.

**Comment by Dr-Bean** [comment]
As I said, the frontend is a piece of cake ;). If you essentially follow the installer, that should work (just as it did the first time...;)). 
Obviously, you can make a backup of the current dir to be safe, but apart from `config.php`, there shouldn't be anything in there that's user-specific. So during our upgrade, we save config.php, delete the dir, put a fresh set of files back, and restore config.php.
The only thing that might be worth mentioning is the .htaccess file, but if you changed that file manually, you'd need to redo those changes anyway.

**Comment by Zeawiel** [comment]
Hello,
As said before, thank for all the work on the repo ;)

Unfortunately, I have the same issue. But with a fresh new install of Owncloud.

I'm using DS415+ / DSM 5.2-5592 Update 3

First I installed MariaDB, actived service web.
I create a folder to store data under /volume1/Stockage/OwnCloud with http R&W rigths.

Then install the package OwnCloud 8.1.1-7 from the package installer. 
The install finish properly, but when i want to access it from myIP/owncloud it says "sorry, the page is not found"
(or "Désolé, la page que vous recherchez est introuvable" in french)

I think this could be related to an update of either DSM or Owncloud package.

**Comment by Dr-Bean** [comment]
@Zeawiel, also for you, I can't reproduce this on latest DSM and latest ownCloud package, same arch. ownCloud is nothing special from a packaging perspective, and unless Apache isn't running, I don't see how DSM would be related.
My guess at this point is that it's something particular to your setup, although I don't have a clue what that would be.

Also for you: If a reinstall of the package does not work, then start with downloading the owncloud file from https://owncloud.org/install/#instructions-server and unzip it to `/volume1/web/owncloudtest` (or whatever dir you choose). You should be able to open ownCloud immediately via http://ip/owncloudtest (or replace with the dirname you chose)

**Comment by Zeawiel** [comment]
Hello,
That was weird. In fact, I do all the job from the outside of my house, connecting through the quickconnect (or the IP/management regular side). I think that the www/owncloud directory is not accessible from outside the LAN.
I registered the NAT to fwd port 80 & 443 to the syno before installed Owc.

Now, from the "Local_IP/owncloud", it seems to work pretty well.

I'm sorry to bothered you with that. I hope now that the last configuration task is made, I'll can access it from outside.

**Comment by ThomasDaheim** [comment]
Hi,

after manually copying my <code>config.php</code> from my old owncloud folder to the new one now also 8.1 works as expected!

Thanks for helping me sorting that out,
Thomas
